graylog2-server/src/main/java/org/graylog2/rest/resources/streams/StreamResource.java

```diff
@@ -233,13 +247,10 @@ public StreamListResponse get() {
     @ApiOperation(value = "Get a list of all enabled streams")
     @Produces(MediaType.APPLICATION_JSON)
     public StreamListResponse getEnabled() throws NotFoundException {
-        final List<Stream> enabledStreams = streamService.loadAllEnabled();
-        final List<Stream> streams = new ArrayList<>(enabledStreams.size());
-        for (Stream stream : enabledStreams) {
-            if (isPermitted(RestPermissions.STREAMS_READ, stream.getId())) {
-                streams.add(stream);
-            }
-        }
+        final List<Stream> streams = streamService.loadAllEnabled()
+                .stream()
+                .filter(stream -> isPermitted(RestPermissions.STREAMS_READ, stream.getId()))
+                .collect(Collectors.toList());
```

Comment:
same as above. Stream.toList() is supposed to be more efficient (no copying etc. as with Collectors)